index.html

```diff
@@ -173,126 +173,364 @@ <h2>
+                    If this request's <a>pointerLockOptions</a> are equivalent to the user agent's current
+                    <a>pointerLockOptions</a> then the <a>pointer-lock target</a>
+                    must be updated to this element, a <a>pointerlockchange</a>
+                    event sent, and the promise must <a>resolve</a>.
+                  </li>
+                  <li>
+                    If this request's <a>pointerLockOptions</a> are not equivalent to the user agent's current
+                    <a>pointerLockOptions</a> and this request's <a>pointerLockOptions</a> are not supported:
+                    <ol>
+                      1. <a href="https://html.spec.whatwg.org/multipage/webappapis.html#queue-an-element-task">
+                      Queue an element task</a> on the 
+                      <a href="https://html.spec.whatwg.org/multipage/webappapis.html#user-interaction-task-source">
+                      User Interaction task source</a>, given this, to perform the following steps:
+                      <ol type='i'>
+                        <li>
+                          <a href=https://dom.spec.whatwg.org/#concept-event-fire">Fire an event</a> named
+                          <a>pointerlockerror</a> at <a>this</a>'s 
+                          <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a>.
+                        </li>
+                        <li>
+                          Return <a>a promise rejected with</a> a "{{NotSupportedError}}" {{DOMException}}.
+                        </li>
+                      </ol>
+                    </ol>
+                  </li>
+                  <li>
+                    If this request's <a>pointerLockOptions</a> are not equivalent to the user agent's current
+                    <a>pointerLockOptions</a> and this request's <a>pointerLockOptions</a> are supported the
+                    user agent's <a>pointer-lock target</a> must be updated to this element, the user agent's
+                    <a>pointerLockOptions</a> must be updated to this request's <a>pointerLockOptions</a>, a 
+                    <a>pointerlockchange</a> event sent, and the promise must <a>resolve</a>.
+                  </li>
+                </ol>
+              </li>
+              <li>
+                If any element (including this one), whose <a>shadow-including
+                root</a> is same as this element's <a>shadow-including root</a>, is
+                already locked (or pending lock) the pointer lock <a>pointer-lock target</a>
+                must be updated to this element, a <a>pointerlockchange</a>
+                event sent and the promise must <a>Resolve</a>.
+              </li>
+              <li>
+                If the request was not started from an <a>engagement gesture</a> and 
+                the Document has not previously released a successful Pointer Lock with 
+                {{Document/exitPointerLock()}}:
```

Comment:
Could this be made precise by checking if the user agent's pointer-lock target is non-null?